crates/jit/src/compiler.rs

```diff
@@ -183,3 +184,24 @@ impl Compiler {
         })
     }
 }
+
+impl Hash for Compiler {
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        // Hash compiler's flags: compilation strategy, isa, frontend config,
+        // misc tunables.
+        self.strategy().hash(hasher);
```

Comment:
Could this use destructuring like:

```rust
let Compiler { field1, field2, } = self;
```
and then hash each individual field? That way we can be sure we're not forgetting anything if the contents change over time.